### DISCUSSION TITLE
v.3.1.0 - New features and US Version

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#299] [Announcements]
### I think we have created a really unique NSPanel version here. And looking on all the features in this new release as well as the speed and effort we put into the development is surely worth a small donation ;) 

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;

### Therefore I would like to thank everyone who is working on the project but also all the users who are helping other users with issues - thank you!

&nbsp;
&nbsp;

#### Before you do the ESPHome update please change the path to the new TFT file. This could look like this

`nextion_update_url: "http://HOME-ASSISTANT-IP:8123/local/nspanel/nspanel_eu.tft" # URL to local tft File`

`nextion_update_url: "http://HOME-ASSISTANT-IP:8123/local/nspanel/nspanel_us.tft" # URL to local tft File`

&nbsp;

> ### ⚠️ If you have issues installing the tft-file, DO NOT create a new isse, read here 
> [Common issues](https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki) / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/166 / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/177 / https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/197
> ### Or use the Issues search: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues?q=is%3Aissue+tft+is%3Aclosed
> ### This is not a bug in this project, this is how the implemetion of nextion in esphome just works

&nbsp;

## Files that need to be reloaded:

1. nspanel_esphome.yaml
2. nspanel_blueprint.yaml
3. nspanel.tft

 _Update procedure can be read here:_

EN: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-HowTo---All-important-thing-you-should-know

DE: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(DE)-HowTo---Alle-wichtigen-Dinge-die-man-wissen-sollte

&nbsp;

## Community Updates
- Simplify and remove useless code in ESP32 - thanks @X-Ryl669
- Work on NSPanel US version - thanks @miniminne  and @Loose-Moose77

&nbsp;

## Overview of all changes

1.  **_"Entity Card Page"_** - 4 new entity card pages with a total of 32 entity fields &nbsp;

2. **_"Icon Font"_** - new icon font with 6896 icons. Can already be used on the entity page - thanks @joBr99  &nbsp;

3. **_"QR Code Page"_** - can be set via Blueprint or HA Service &nbsp;

5. **_"Relay Fallback"_** - Relays can be controlled with the hardware buttons if the NSPanel loses connection to WiFi or HA &nbsp;

6. **_"Page Time Out"_** - Time to return to the home page &nbsp;

7. **_"New Boot Page Design"_** - Display of Blueprint, Esphome and TFT version as well as the IP address of the NSPanel &nbsp;

8. **_"add "Long-Press" on the hardware / physical keys"_** - with a long press on the hardware button opens e.g. light settings &nbsp;

9. **_"Changes from the Blueprint are applied immediately"_** - After changes in the Blueprint, a restart of the NSPanel is no longer necessary. &nbsp;

10. **_"NSPanel US Version"_** - first official TFT version for the NSPanel US in portrait mode 

11. **_"Bug fixes and improvements in the blueprint, ESPHome, and TFT code."_** 

&nbsp;

### 1. Entity Card Page
We have added 4 new entity card pages. Each page has space for 8 entities, so in total there are now 32 entities at your disposal.
The entities can now be selected via the blueprint. 

#### Additionally there is the possibility to choose an icon of your choice from 6896 icons.

Of course there is also the possibility to enter your own entity label and to assign a label for the 4 pages. If no entity label is set, the friendly name is taken from HA.
As you are already used to, we have again implemented the navigation with swipe to the right, left, up and down.

Blueprint:

Activate Entity Pages:
![Bildschirmfoto 2023-02-03 um 23 15 30](https://user-images.githubusercontent.com/41958506/216721185-5f3ec1a5-ca17-4d23-838b-bd5ec2c516e6.png)

Entities:
![Bildschirmfoto 2023-02-03 um 23 11 57](https://user-images.githubusercontent.com/41958506/216720844-7d980a0f-bfe3-4892-ad75-ade983d14b85.png)

Nspanel:
![Entity Page](https://user-images.githubusercontent.com/41958506/215906572-a447a81b-179b-4c49-b4e5-9b0b2f594c16.png)

&nbsp;

### 2. Icon Font
We have added a new font that contains 6896 Material Design icons (the same as HA uses).
The icons can currently be set in the blueprint for the entities.
**To use the icons in the Blueprint you would have to click on the "U" on the website to copy the icon and then paste it in the Blueprint** _(Attention: Only a "strange" character  is displayed, but that is correct)._

Icon Page: https://htmlpreview.github.io/?https://github.com/jobr99/Generate-HASP-Fonts/blob/master/cheatsheet.html

![Bildschirmfoto 2023-01-31 um 23 24 24](https://user-images.githubusercontent.com/41958506/215897637-e515ec32-3849-483b-927c-9d7e15b939b8.png)

![Bildschirmfoto 2023-01-31 um 23 27 43](https://user-images.githubusercontent.com/41958506/215898075-fe538cf5-85a3-4f83-b3fc-fbc8c20f2498.png)

&nbsp;

### 3. QR Code Page
QR Code can be specified and set in the Blueprint. However, it is also possible to send a QR code via HA Services. If the QR code is sent via HA Services, the QR code page opens automatically.
If the QR Code page is opened again via the NSPanel, the QR Code will be overwritten with the value from the Blueprint.

In order for the QR code character to be displayed on the Home Page, it must be activated in the Blueprint before
![Bildschirmfoto 2023-02-03 um 23 28 41](https://user-images.githubusercontent.com/41958506/216723149-835e8077-8c2e-49fa-aaf9-4ceab6778e45.png)

For the QR Code Content you have 50 characters at your disposal which you can use freely.
![Bildschirmfoto 2023-01-31 um 23 36 05](https://user-images.githubusercontent.com/41958506/215899355-e1d04b05-cd71-4c6e-8c5e-264d50e7e43c.png)

HA Service:
![Bildschirmfoto 2023-01-31 um 23 35 37](https://user-images.githubusercontent.com/41958506/215899395-e4aba3d7-e7be-4ba3-954b-027023e90a3f.png)

NSPanel
![QR_Code](https://user-images.githubusercontent.com/41958506/215906530-a9e5d554-82b3-4841-9e06-767df5ba2b62.png)

&nbsp;

### 4. Relay Fallback
with version 3.1.0 we have implemented a Fallback for the relay.
This means that if the NSPanel loses the connection to Home Assistant or WiFi, the automatic fallback is triggered.

When the fallback is activated, the left or right hardware button can be used to manually control relay 1 or relay 2, regardless of how the buttons were previously assigned.

This function must be activated in the Blueprint for both keys

![Bildschirmfoto 2023-02-03 um 23 35 30](https://user-images.githubusercontent.com/41958506/216724093-21fc536e-12bb-443f-9107-680f6c5160d1.png)

![Bildschirmfoto 2023-02-03 um 23 35 39](https://user-images.githubusercontent.com/41958506/216724118-6d303761-efdd-4c45-a378-d68f1d9e291d.png)

&nbsp;

### 5. Page Time Out
Page Time Out means the time until the NSPanel automatically jumps back to the Home Page. 
This time can now be set under HA -> Devices -> NSPanel

If the time is set to 0, jumping back to the home page is deactivated and you can only jump back to the home page via the "X".

This setting applies to all pages.

![Bildschirmfoto 2023-02-03 um 23 47 15](https://user-images.githubusercontent.com/41958506/216725503-b5cfc924-b461-49a3-bc02-830d9348eb6a.png)

&nbsp;

### 6. New Boot Page Design
We have customized the boat page to display more useful information.

The IP address is now displayed at the top of the page.

You can also see the installed TFT, Blueprint and ESPhome version. This should help to get a better overview which versions are installed.

![Boot](https://user-images.githubusercontent.com/41958506/215906839-23f407c0-2ec2-4dd9-91de-a8971fa94b00.png)

&nbsp;

### 7. add "Long-Press" on the hardware / physical keys
We have now added a function that allows you to perform an additional function by a long press on the hardware button.

This could look like this:
On the hardware button, a light has been stored as an entity in the Blueprint. With a long press, the light setting of the stored light entity is now opened.

Another example would be with a cover / shutter. If a cover / shutter entity is stored on the second hardware button, a long press opens the shutter settings.

In short - the function is now the same as for the other 32 virtual buttons .

&nbsp;

### 8. Changes from the Blueprint are applied immediately

Previously, every change in the blueprint required a restart of the NSPanel.

Now, this is no longer necessary. If something is changed in the blueprint and saved, it is immediately adopted. Depending on the changes, it may take a moment to load everything.

The NSPanel does not go into a reboot.
This has the advantage that updates to the blueprint do not disturb people at night or in other situations.

&nbsp;

### 9. NSPanel US Version
Many have been waiting for it and now it's finally here - we have our first NSPanel US version in portrait mode.

Thanks to the great work of @miniminne  and @Loose-Moose77, I was able to complete and finish the topic.

I have rebuilt all pages and designed new elements so that it looks exactly as neat and the same as on the EU version.

I am totally thrilled with the NSPanel US version. The only thing missing is the "Wave" Background, which I will still provide.

&nbsp;

![1  home](https://user-images.githubusercontent.com/41958506/216813036-82508a13-d59a-432f-a533-9fb19c98763f.png) ![2  Button](https://user-images.githubusercontent.com/41958506/216813048-29758424-ce90-4eec-8dea-fdbfa8879b47.png)

![3  Weather](https://user-images.githubusercontent.com/41958506/216813055-5420e673-6f87-4470-9e35-16f99fe499d8.png) ![4  Climate](https://user-images.githubusercontent.com/41958506/216813063-06dce76f-f532-474f-b1da-6b94366e38fa.png)

![5  Brightness](https://user-images.githubusercontent.com/41958506/216813135-a2b47013-6cb6-4554-95f0-efa7a486e7a1.png) ![6  Tempcolor](https://user-images.githubusercontent.com/41958506/216813138-45d4889b-3e6a-4b66-ae9c-53dfcd765077.png)

![7  Colorweehl](https://user-images.githubusercontent.com/41958506/216813144-a027f873-de58-46df-8907-831e3dfd3d6c.png) ![8  Cover](https://user-images.githubusercontent.com/41958506/216813146-58127828-85a4-46f3-b854-ddfb2c923311.png)

![9  Enitity](https://user-images.githubusercontent.com/41958506/216813154-a15f675d-3c48-4bd6-bc2b-d39537e46690.png) ![10  Notification](https://user-images.githubusercontent.com/41958506/216813156-26186a44-498e-40dc-90f7-28040e33097f.png)

![11  QR Code](https://user-images.githubusercontent.com/41958506/216813169-85aa9a24-4223-49e0-98a6-98aa11a78848.png) ![12  Boot](https://user-images.githubusercontent.com/41958506/216813176-fd016a42-0d30-44f8-8def-525a94e53a79.png)

![13  Settings](https://user-images.githubusercontent.com/41958506/216813185-a6238808-7df2-4fa3-8015-2fdff39e147a.png)

&nbsp;

### 10. Bug fixes and improvements in the blueprint, ESPHome, and TFT code.
As with every release, we have made some adjustments and fixes to the code to provide a better experience with our NSPanel version.

&nbsp;
&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

## What's Changed
* Update README.md by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/194
* Update Readme by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/195
* change tft baud to bauds by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/198
* change tft baud to bauds by @Blackymas in https://github.com/Blackymas/NSPanel_HA_Blueprint/pull/199

**Full Changelog**: https://github.com/Blackymas/NSPanel_HA_Blueprint/compare/v.3.0.0...v.3.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.1.0'>v.3.1.0 - New features and US Version</a>.</em>

**Comment by pwirkus** [comment]
![20230205_131807](https://user-images.githubusercontent.com/118907721/216823394-b2a47df7-54b0-4679-9715-246896cf47de.jpg)

does this mean that the update from version 3.0 to version 3.1 was performed correctly?

**Reply by Blackymas** [reply]
Jep

**Comment by pwirkus** [comment]
please tell me how to add alarm panel or media panel? is this option not available yet?

**Reply by Blackymas** [reply]
I hope the question was meant funny;) 
In the release notes are all new features and improvements in it. Mediaplayer is still in development and will probably come in the next release.

**Reply by pwirkus** [reply]
maybe it's funny :) but I prefer to ask before I break something :) I care more about the alarm panel with the possibility of arming and disarming with a code as it is in the nspanel lovelace version :) but I prefer the blue version than tasmota, when would be the approximate time of implementing the alarm panel? thank you for your work and best regards

**Reply by Blackymas** [reply]
both topics are actively being worked on - it is hard to say how fast we are progressing and if problems are occurring - but we are working on it

**Reply by pwirkus** [reply]
ok , thank you

**Comment by smartmatic** [comment]
Thanks @Blackymas for this awesome update! I've performed it and works very well! YT video will be released soon ;-)
But one thing: How can i delete the old tft from the panel to save disk space?

**Reply by Blackymas** [reply]
when you upload a new tft, the old one is replaced. nothing needs to be deleted.

ps I am excited and look forward to it ;)

**Comment by Stasky1968** [comment]
Great job , do you are working also on US panel landscape ? Is similar of EU version only the two fisic button are on the right

**Reply by Blackymas** [reply]
Try it with the eh version. Some users have reported that it works. You can find the posts if you use the issues search

**Comment by Leeroy-J** [comment]
**THANK YOU!** Playing with it now, and so far, so good.

**Reply by JUK-Bavaria** [reply]
Great update! Is there any possibility to replace the output of an entity?
"on" and "off" are not so perfect for a window or a door. Fine would be a colored icon or open/closed.
